docs/guide/uploading-downloading-binary-content.md

```diff
@@ -0,0 +1,147 @@
+
+```js
+// define our app and dependencies (remember to include firebase!)
+angular
+  .module("sampleApp", [
+    "firebase"
+  ])
+  .controller("SampleCtrl", SampleCtrl);
+
+// inject $firebaseStorage into our controller
+function SampleCtrl($firebaseStorage) {
+  // create a Firebase Storage Reference for the $firebaseStorage binding
+  var storageRef = firebase.storage().ref('userProfiles/physicsmarie');
+  var storage = $firebaseStorage(storageRef);
+}
+SampleCtrl.$inject = ["$firebaseStorage"];
+```
+
+## API Summary
+
+The Firebase Storage service is created with several special $ methods, all of which are listed in the following table:
+
+| Method  | Description |
+| ------------- | ------------- |
+| [`$put(file, metadata)`](/docs/reference.md#putfile-metadata) |	Uploads file to configured path with optional metadata. Returns an AngularFire wrapped UploadTask. |
+| [`$putString(string, format, metadata)`](/docs/reference.md#putstringstring-format-metadata)	| Uploads a upload a raw, base64, or base64url encoded string with optional metadata. Returns an AngularFire wrapped UploadTask. |
+| [`$getDownloadURL()`](/docs/reference.md#getdownloadurl) |	Returns a promise of the download URL for the file stored at the configured path. |
+| [`$getMetadata()`](/docs/reference.md#getmetadata) | Returns a promise of the metadata of the file stored at the configured path. |
+| [`$updateMetadata(metadata)`](/docs/reference.md#updatemetadatametadata) | Updates the metadata of the file stored at the configured path. Returns a promise containing the complete metadata or en error. |
+| [`$delete()`](/docs/reference.md#delete) | Permanently deletes the file stored at the configured path. Returns a promise that is resolved when the delete completes. |
+| [`$toString()`](/docs/reference.md#tostring) | Returns a string version of the bucket path stored as a `gs://` scheme. |
+
+
+## Uploading files
+To upload files use either the `$put()` or `$putString()` methods. These methods
+return an `UploadTask` which is wrapped by AngularFire to handle the `$digest` loop.
+
+```js
+function SampleCtrl($firebaseStorage) {
+  // create a Firebase Storage Reference for the $firebaseStorage binding
+  var storageRef = firebase.storage().ref('userProfiles/physicsmarie');
+  var storage = $firebaseStorage(storageRef);
+  var file = // get a file from the template (see Retrieving files from template section below)
+  var uploadTask = storage.$put(file);
+  // of upload via a RAW, base64, or base64url string
+  var stringUploadTask = storage.$putString('5b6p5Y+344GX44G+44GX44Gf77yB44GK44KB44Gn44Go44GG77yB', 'base64');
+}
+SampleCtrl.$inject = ["$firebaseStorage"];
+```
+
+### Upload Task API Summary
+
+| Method  | Description |
+| ------------- | ------------- |
+| [`$progress(callback)`](/docs/reference.md#uploadtask-progress) |	Calls the provided callback function whenever there is an update in the progress of the file uploading. |
```

Comment:
Links for all of these methods are broken. It looks like they don't need a leading "uploadtask-".